Don't overwrite serialVersionUID if already defined in the class

fviale opened the pull request.

fviale committed:
Don't overwrite existing serialVersionUIDs (as they can be fixed for a reason).

fviale committed:
changed version number

laurentpellegrino commented:
@fviale I understand the need but there are some classes, even in _scheduling_ project (for instance `JobResultImpl`, `JobStateImpl`, etc.), which define a _serialVersionUID_ field. If the semantic of the plugin changes, then it would be more safe to remove all existing _serialVersionUID_ field definitions from projects that are released. Otherwise, in the future it is possible that a manually set value for a new release conflicts with one already used for a previously release. In that case it could be really tricky to understand the root cause.

fviale commented:
Ah, I actually searched in the scheduling and programming projects but had a wrong filter put, I'm going to look at those ids. Indeed it would be risky. Maybe a custom variable could be defined in the class and checked by the plugin (not very standard I know)

fviale committed:
Added a configurable overwrite behavior, true by default.

fviale committed:
Added a configurable forceUIDOnThrowable behavior :
- this setting force the serialUID to be 1L for all Throwable classes
- works in complement with the overwrite=true setting. In that case, the forceUIDOnThrowable has higher priority.

laurentpellegrino commented:
:+1:
